test/browser/fixture.test.js

```diff
@@ -0,0 +1,216 @@
+import { focusWithKeyboard } from '../../src/browser/focus.js';
+import { sendMouse } from '@web/test-runner-commands';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+
+const resolves = new Map();
+
+const slowElem = defineCE(
+	class extends LitElement {
+		static get properties() {
+			return { nested: { type: Boolean } };
+		}
+		constructor() {
+			super();
+			this.nested = false;
+			this.finished = false;
+			this.promise = new Promise((resolve) => resolves.set(this.id, resolve));
+			this.promise.then(() => this.finished = true);
+		}
+		render() {
+			return this.nested ? html`${unsafeHTML(`<${slowElem} id="nested"></${slowElem}>`)}` : nothing;
+		}
+		async getUpdateComplete() {
+			await super.getUpdateComplete;
+			return this.promise;
+		}
+	}
+);
+
+describe('fixture', () => {
+
+	afterEach(() => restore());
+
+	describe('reset', () => {
+
+		const mousePos = { x: 0, y: 0 };
+		function onMouseMove(e) {
+			mousePos.x = e.clientX;
+			mousePos.y = e.clientY;
+		}
+
+		beforeEach(() => {
+			window.addEventListener('mousemove', onMouseMove);
+		});
+
+		afterEach(() => {
+			window.removeEventListener('mousemove', onMouseMove);
+		});
+
+		it('should reset window scroll position', async() => {
+			const elem = html`<div style="height: 2000px; width: 2000px;">big thing</div>`;
+			await fixture(elem);
+			window.scrollTo(200, 300);
+			expect(window.scrollX).to.equal(200);
+			expect(window.scrollY).to.equal(300);
+			await fixture(elem);
+			expect(window.scrollX).to.equal(0);
+			expect(window.scrollY).to.equal(0);
+		});
+
+		it('should reset mouse position', async() => {
+			const elem = html`<p>hello</p>`;
+			await fixture(elem);
+			await sendMouse({ type: 'move', position: [5, 10] });
+			expect(mousePos.x).to.equal(5);
+			expect(mousePos.y).to.equal(10);
+			await fixture(elem);
+			expect(mousePos.x).to.equal(0);
+			expect(mousePos.y).to.equal(0);
+		});
+
+		it('should reset focus', async() => {
+			const elem = await fixture(html`<button>hello</button>`);
+			expect(document.activeElement).to.equal(document.body);
+			await focusWithKeyboard(elem);
+			expect(document.activeElement).to.equal(elem);
+			await fixture(html`<div>hello</div>`);
+			expect(document.activeElement).to.equal(document.body);
+		});
+
+		it('should reset text direction', async() => {
+			await fixture(html`<p>hello</p>`);
+			expect(document.documentElement.hasAttribute('dir')).to.be.false;
+			await fixture(html`<p>mrhban</p>`, { rtl: true });
+			expect(document.documentElement.getAttribute('dir')).to.equal('rtl');
+			await fixture(html`<p>hello</p>`);
+			expect(document.documentElement.hasAttribute('dir')).to.be.false;
+		});
+
+		it('should use RTL text direction with AR language', async() => {
+			await fixture(html`<p>mrhban</p>`, { lang: 'ar' });
+			expect(document.documentElement.getAttribute('dir')).to.equal('rtl');
+		});
+
+		it('should default language to EN', async() => {
```

Comment:
Should we remove the `lang` attribute first since this is effectively doing the same thing as the test below?
```suggestion
		it('should default language to EN', async() => {
			document.documentElement.removeAttribute('lang');
			expect(document.documentElement.getAttribute('lang')).to.be.null;
```